### DISCUSSION TITLE
How to skip a dynamic node?

### DISCUSSION THREAD

**Discussion by itsramiel** [L3MON4D3/LuaSnip#1159] [Q&A]
Hello 👋

I have the following snippet:
```
s(
  "f",
  fmt(
    [[
  {}function {}({}){} {{
    {}
  }}
  ]],
    {
      c(1, {
        t("export "),
        t(""),
      }),
      d(2, function(_, parent)
        return sn(nil, {
          i(1, parent.env.TM_FILENAME_BASE),
        })
      end),
      i(3),
      i(4),
      i(0),
    }
  )
)
```

Assming I am in a `add.ts` file, it produces the following snippet:

<img width="235" alt="image" src="https://github.com/L3MON4D3/LuaSnip/assets/80689446/3697617b-bc56-451c-9fbd-dd8c0af81391">

I have an issue when I am at the dynamic node here:
<img width="220" alt="image" src="https://github.com/L3MON4D3/LuaSnip/assets/80689446/dcfacd3e-4606-4df3-9007-ef23946e70ee">

If I go ahead and type something else for the function name then I can move on to the next node, but if dont change it as keep the dynamic node as `add` then going to the next node does not work and I am stuck at the dynamic node.

Am i doing something wrong or how can I fix it?

**Comment by L3MON4D3** [comment]
Hey :)
I think something in your setup is wrong/different from mine, because everything seems to work correctly on my machine

https://github.com/L3MON4D3/LuaSnip/assets/41961280/5277fd0a-da75-4e0e-9889-df86a873583e

Could you try reducing your setup (remove luasnip-configuration/other plugins) until the issue disappears?

**Comment by itsramiel** [comment]
So I experienced that this happens only sometimes. Now I exited neovim and started it again and it works fine. I am not really sure how to reproduce it consistently.

Here is my setup btw:
```
return {
	"L3MON4D3/LuaSnip",
	-- follow latest release.
	version = "v2.*", -- Replace <CurrentMajor> by the latest released major (first number of latest release)
	-- install jsregexp (optional!).
	build = "make install_jsregexp",
	config = function()
		local ls = require("luasnip")

		ls.setup({
			update_events = { "TextChanged", "TextChangedI" },
		})

		vim.keymap.set({ "i" }, "<C-K>", function()
			ls.expand()
		end, { silent = true })

		vim.keymap.set({ "i", "s" }, "<C-L>", function()
			ls.jump(1)
		end, { silent = true })

		vim.keymap.set({ "i", "s" }, "<C-J>", function()
			ls.jump(-1)
		end, { silent = true })

		vim.keymap.set({ "i", "s" }, "<C-E>", function()
			if ls.choice_active() then
				ls.change_choice(1)
			end
		end, { silent = true })

		require("luasnip.loaders.from_lua").load({ paths = "~/.config/nvim/lua/snippets" })

		ls.filetype_extend("javascriptreact", { "javascript" })
		ls.filetype_extend("typescriptreact", { "javascript" })
		ls.filetype_extend("typescript", { "javascript" })
	end,
}
```

I think it is pretty simple.

I can keep an eye on it and try to figure out when it happens

**Reply by L3MON4D3** [reply]
Hmm, yeah that looks very innocuous..
Not being able to jump sounds an awful lot like the snippet is getting removed completely (not the text, but the object that enables the jumps).. is it possible you have some kind of formatter enabled? This would also explain why changing the text causes the removal

**Reply by L3MON4D3** [reply]
Ah, wait, wrong way around, not changing causes the removal somehow.. then I'm not sure again :sweat_smile:

**Reply by itsramiel** [reply]
Here is the behavior I experience if this helps out.

When I am at the dynamic node and cant jump to the next node, if I type something to change the dynamic node, then I am able to go to the next node. Also if I am at the dynamic node and press escape to go to normal mode and then back to insert mode I can press the next node key binding and I continue after the dynamic node.

Not sure if that helps but just in case, but again I will keep an eye and try to find out how to consistently reproduce

**Comment by itsramiel** [comment]
I updated to neovim stable 0.10 and seems to not happen anymore
